### DISCUSSION TITLE
Unhandled Exception: SIGSEGV: invalid attempt to write memory at address 0x0000000000000003

### DISCUSSION THREAD

**Discussion by switchyfoot** [wolveix/satisfactory-server#195] [Q&A] [answered]
Been getting server crashes with this error:
Unhandled Exception: SIGSEGV: invalid attempt to write memory at address 0x0000000000000003

Googling shows this problem was related back in update 5 with ipv6 or something. In my case its not preventing start, it just crops up every few days. I do have two NICs, so maybe adding the -MultiHome=123.123.123.123 (this would be my IP) may fix it. I just don't know where to add this?

**Comment by wolveix** [comment]
Hey @switchyfoot, could you send your full log? It's very unlikely to be an IP binding issue, as we addressed that for the image at the time :)

**Comment by switchyfoot** [comment]
[FactoryGame-backup-2023.07.11-23.44.17.log](https://github.com/wolveix/satisfactory-server/files/12028064/FactoryGame-backup-2023.07.11-23.44.17.log)

I have a few others that have are from crashes but they are fairly large files let me know if i should post more.

**Reply by wolveix** [reply, marked as answer]
@switchyfoot here is the relevant part:

```
[2023.07.11-23.44.17:489][693]LogGame: Warning: BP_MaterialEffect_Dismantle_C_2147200765::SetMeshes, This cannot be called after PreStarted.
[2023.07.11-23.44.17:963][707]LogCore: Error: appError called: Assertion failed: (Index >= 0) & (Index < ArrayNum) [File:Runtime/Core/Public\Containers/Array.h] [Line: 674] 
Array index out of bounds: 233 from an array of size 233

[2023.07.11-23.44.17:968][707]LogCore: === Critical error: ===
Unhandled Exception: SIGSEGV: invalid attempt to write memory at address 0x0000000000000003

[2023.07.11-23.44.17:968][707]LogCore: Assertion failed: (Index >= 0) & (Index < ArrayNum) [File:Runtime/Core/Public\Containers/Array.h] [Line: 674] 
Array index out of bounds: 233 from an array of size 233
```

This is a game bug, I'd recommend checking the Coffee Stain community forum to see whether it's already been reported (and if not, report it yourself). It's likely that you're experiencing some kind of edge case in your game save that's triggering the seg fault.

**Reply by switchyfoot** [reply]
Thank you for your help and for this amazing container. I will make a post on the developer forums and link to it here when I do.

Is there a way to have the container auto restart when this error happens?

**Reply by wolveix** [reply]
@switchyfoot you're very welcome! Of course :) Assuming that the container is stopping, you just need to set the container to auto restart (which we recommend by default with the provided Docker Compose sample in the `README`. If you're using `docker run` instead (which I'd recommend against for the sake of easy updating and such), you can add `--restart unless-stopped` :)

**Comment by shaunVorster** [comment]
Good evening @wolveix.

Firstly, thanks for the way server container.

I have also seen this issue specifically on version 1.5.5 of the container. Rolling back to v 1.5.3 seems to solve it.
In my case I can't even get v1.5.5 to start.
The container spins up and then restarts after roughly a minute.

V1.5.3 has no issues at all

**Reply by wolveix** [reply]
@shaunVorster appreciate the kind words. Please set `SERVERIP=""` and let me know if that makes a difference for you. If not, please open a new issue with the requested template information. Thanks!

**Reply by shaunVorster** [reply]
Thanks for the quick response.

I will climb into it tomorrow and let you know.
That can be set in the compse Environment variables, correct? 
Thanks very much

**Reply by wolveix** [reply]
@shaunVorster you're welcome! Yep, it's just one of the environment variables. Emptying it should disable `-multihome`, which should resolve your issue

**Reply by shaunVorster** [reply]
Good morning @wolveix!
Seems like that managed to solve the startup issue on v1.5.5. 
I will keep testing but for now it seems to be working perfectly.

Thanks very much
